src/launcher/plugin.ts

```diff
@@ -0,0 +1,214 @@
+
+import {
+  IPathTracker
+} from '../filebrowser';
+
+import {
+  IServiceManager
+} from '../services';
+
+/**
+ * The landing page extension.
+ */
+export
+const launcherExtension: JupyterLabPlugin<void> = {
+  id: 'jupyter.extensions.launcher',
+  requires: [IServiceManager, IPathTracker, ICommandPalette],
+  activate: activateLauncher,
+  autoStart: true
+};
+
+
+
+function activateLauncher(app: JupyterLab, services: IServiceManager, pathTracker: IPathTracker, palette: ICommandPalette): void {
+  let launcherModel = new LauncherModel(pathTracker);
+  let launcherWidget = new LauncherWidget()
+
+  launcherWidget.model = launcherModel;
+  launcherWidget.id = 'landing-jupyterlab-widget';
+  launcherWidget.title.label = 'Launcher';
+  launcherWidget.addClass('jp-Launcher');
+
+  // hardcoded defaults
+  let names = [
+      'Notebook',
+      'Console',
+      'Terminal',
+      'Text Editor',
+  ];
+
+  let actions = [
+      'file-operations:new-notebook',
+      `console:create-${services.kernelspecs.default}`,
+      'terminal:create-new',
+      'file-operations:new-text-file',
+  ]
+
+  let list : IDisposable[] = []; //DEMO
+  let l : IDisposable; //DEMO
+
+  for (let i in names) {
+      let itemName = names[i];
+      let action = actions[i];
+      l = launcherModel.add(itemName, () => app.commands.execute(action, void 0));
+      list.push(l) // DEMO
+  }
+
+  // DEMO
+  launcherModel.add("Add Random", () => {
+
+      let index = Math.floor(Math.random() * actions.length);
+      let itemName = names[index];
+      let action = actions[index];
+      l = launcherModel.add(itemName, () => app.commands.execute(action, void 0));
+      list.push(l);
+  });
+
+  launcherModel.add("Remove Last", () => {
+      list.pop().dispose();
+  });
+  //  end DEMO
+
+
+
+  app.commands.addCommand('jupyterlab-launcher:add-item', {
+    label: 'Add Launcher Item',
+    execute: args => {
+        // I'm not sure how to handle this -pi
+       // launcherModel.add(args['name'] as string, args['clickCallback']);
+    }
+  });
+
+
+  app.commands.addCommand('jupyterlab-launcher:show', {
+    label: 'Show Launcher',
+    execute: () => {
+      if (!launcherWidget.isAttached) {
+        app.shell.addToLeftArea(launcherWidget);
+      } else {
+          app.shell.activateMain(launcherWidget.id);
+      }
+    }
+  });
+
+  palette.addItem({
+    command: 'jupyterlab-launcher:show',
+    category: 'Help'
+  });
+
+  app.shell.addToLeftArea(launcherWidget);
+}
+
+
+class LauncherItem {
+    constructor(public name: string, public clickCallback: () => void) {
+    }
+}
+
+
+/**
+ * LauncherModel keeps track of the path to working directory and has a list of
+ * LauncherItems.
+ *
+ */
+class LauncherModel  extends VDomModel {
+    items : LauncherItem[] = [];
+    constructor(public pathTracker : IPathTracker) {
+        super();
+
+        // ensure we re-render on changes to the working directory
+        pathTracker.pathChanged.connect(() => {
```

Comment:
I don't feel great about passing the path tracker into the launcher model, but I don't know of a better pattern for something like this.